addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -639,3 +650,25 @@ def unassign(self):
             .mapped('picking_ids')\
             .filtered(lambda sp: sp.state not in ('done', 'cancel'))\
             .write({'batch_id': False})
+
+    def remove_pickings(self):
+        """
+        Remove pickings from batch if they are not started
+        Backorder half-finished pickings
+        """
+        Picking = self.env['stock.picking']
+
+        pickings_to_remove = Picking.browse()
+        pickings_to_add = Picking.browse()
+
+        for picking in self.picking_ids:
+            started_lines = picking.mapped('move_line_ids').filtered(lambda x: x.qty_done > 0)
+            if started_lines:
+                # backorder incomplete moves
+                if picking._requires_backorder(started_lines):
+                    pickings_to_add += picking._backorder_movelines(started_lines)
```

Comment:
Might need to add context variable to lock batch status update.

Reply:
Same for line 673.
Then in line 674, don't lock the status, so it is properly updated.